Sync/Synchronizers/ClientsSynchronizer.swift

```diff
@@ -111,6 +111,7 @@ open class ClientsSynchronizer: TimestampedSingleCollectionSynchronizer, Synchro
     }
 
     var localClients: RemoteClientsAndTabs?
+    var hasLocalClientIdChanged: Bool = false
```

Comment:
Instead of this new bool, it looks like you could instead set `clientRecordLastUpload` to zero, which will force `maybeUploadOurRecord` to upload it.

Reply:
I thought of that, but I didn't want to overload the meaning of that variable. I _think_ (or maybe) hope having this bool makes it easier to understand what's going on, but I'm open to changes.

Reply:
I don't really care - if you do stick with a separate variable, consider naming it something like "blahMigrated", to help make it clear it's a one-off and not an expected situation. OTOH though, I still don't hate overloading `clientRecordLastUploaded` as where it's used, it talks about "is the client record stale", and in this situation it is reasonable to describe it as "stale".

But whatever :)